spec.html

```diff
@@ -14875,16 +14875,19 @@ <h1>
         <emu-alg>
           1. Let _O_ be _taRecord_.[[Object]].
           1. Let _bufferByteLength_ be _taRecord_.[[CachedBufferByteLength]].
-          1. Assert: IsDetachedBuffer(_O_.[[ViewedArrayBuffer]]) is *true* if and only if _bufferByteLength_ is ~detached~.
-          1. If _bufferByteLength_ is ~detached~, return *true*.
+          1. If IsDetachedBuffer(_O_.[[ViewedArrayBuffer]]) is *true*, then
+            1. Assert: _bufferByteLength_ is ~detached~.
+            1. Return *true*.
+          1. Assert: _bufferByteLength_ is a non-negative integer.
           1. Let _byteOffsetStart_ be _O_.[[ByteOffset]].
           1. If _O_.[[ArrayLength]] is ~auto~, then
             1. Let _byteOffsetEnd_ be _bufferByteLength_.
           1. Else,
             1. Let _elementSize_ be TypedArrayElementSize(_O_).
-            1. Let _byteOffsetEnd_ be _byteOffsetStart_ + _O_.[[ArrayLength]] × _elementSize_.
+            1. Let _arrayByteLength_ be _O_.[[ArrayLength]] × _elementSize_.
+            1. Let _byteOffsetEnd_ be _byteOffsetStart_ + _arrayByteLength_.
+          1. NOTE: A 0-length TypedArray with [[ByteOffset]] of _bufferByteLength_ is not considered out-of-bounds.
```

Comment:
```suggestion
          1. NOTE: A 0-length TypedArray whose [[ByteOffset]] is _bufferByteLength_ is not considered out-of-bounds.
```

Same below. This matches the phrasing we used in https://github.com/tc39/ecma262/pull/3396/files#diff-181371b08d71216599b0acccbaabd03c306da6de142ea6275c2135810999805aR49765.

Reply:
Done.